crates/miden-lib/src/transaction/mod.rs

```diff
@@ -199,11 +200,14 @@ impl TransactionKernel {
     ///   account.
     pub fn extend_advice_inputs_for_account(
         advice_inputs: &mut AdviceInputs,
-        account_header: &AccountHeader,
-        account_code: &AccountCode,
-        storage_header: &AccountStorageHeader,
-        merkle_path: &MerklePath,
+        foreign_account_inputs: &ForeignAccountInputs,
     ) -> Result<(), MerkleError> {
```

Comment:
I think this is fine for now, but in the future, this could be more generic to handle loading of advice inputs for native  accounts as well. We'll probably make this change when we move to lazy account loading.